### PR TITLE
Add User-Agent and API key warnings

### PR DESCRIPTION
**Pull request by cpcurtis1218**

### CODE REVIEW

File: lib/klaviyo/client.rb [hunk 1]
```diff
@@ -44,12 +46,13 @@ def self.public_request(method, path, **kwargs)
       if method == HTTP_GET
         params = build_params(kwargs)
         url = "#{BASE_API_URL}/#{path}?#{params}"
-        res = Faraday.get(url).body
+        res = Faraday.get(url, {}, { 'User-Agent' => KL_USER_AGENT }).body
```
Comment (remstone7) [inline lib/klaviyo/client.rb:49]: fwiw, cpi;d we make this user agent for all requests as well?

Comment (remstone7) [reply lib/klaviyo/client.rb:49]: nvm it's there

File: klaviyo.gemspec [hunk 2]
```diff
@@ -2,8 +2,8 @@ files = ['klaviyo.gemspec', '{lib}/**/**/*'].map {|f| Dir[f]}.flatten
 
 Gem::Specification.new do |s|
   s.name        = 'klaviyo'
-  s.version     = '2.0.9'
-  s.date        = '2021-07-30'
+  s.version     = '2.1.0'
+  s.date        = '2021-10-01'
```
Comment (remstone7) [inline klaviyo.gemspec:6]: is it possible to make a constant somewhere and then import it to the date? that way we don't need to update both the version and the user agent every release?

Comment (cpcurtis1218) [reply klaviyo.gemspec:6]: I haven't found a good way to do this, the gemspec file is separate from the gem itself so I'm not sure if I can share variables between the two.  I found one way that involved another gem but even then I wasn't able to get it working.

Comment (cpcurtis1218) [reply klaviyo.gemspec:6]: I did change up the variable structure a little to make it clearer though

**Review by njparadis — APPROVED** [review]

**Pull request merged**